### ISSUE TITLE
TypeError: Cannot read property 'annotations' of undefined

### ISSUE THREAD

**Issue by reutopiaer** [angular/quickstart#12]
f1ef6a8b903aad00a2efb607c09ec716ef07b27f  error:
Uncaught (in promise) TypeError: Cannot read property 'annotations' of undefined {stack: (...), message: "Cannot read property 'annotations' of undefined"}

a5f4882ece790b420876e3daae2687405145d8f4  work fine

**Comment by reutopiaer** [comment]
sorry,this is bug report for es6-shim not quickstart

**Comment by reutopiaer** [comment]
es6-shim.js:1 Uncaught RangeError: Maximum call stack size exceeded
    <!doctype html>
    <html>
        <head>
             <title>I</title>
             <script src="quickstart/dist/es6-shim.js"></script>
             <script src="bower_components/webcomponents.js/webcomponents.js"></script>  
       </head>
       <body>
       </body>
    </html>

**Comment by gattoo** [comment]
Same case for me while trying to build a todoApp.

```
TypeError: Cannot read property 'annotations' of undefined

TypeError: Cannot read property 'annotations' of undefined
    at ReflectionCapabilities.execute.$__export.annotations (http://localhost:8080/angular2/src/reflection/reflection_capabilities.js:83:40)
    at Reflector.execute.$__export.annotations (http://localhost:8080/angular2/src/reflection/reflector.js:81:50)
    at DirectiveMetadataReader.execute.$__export.read (http://localhost:8080/angular2/src/core/compiler/directive_metadata_reader.js:33:41)
    at http://localhost:8080/angular2/src/core/compiler/compiler.js:127:35
    at Array.map (native)
    at Function.execute.$__export.map (http://localhost:8080/angular2/src/facade/collection.js:178:26)
    at Compiler.execute.$__export.createSteps (http://localhost:8080/angular2/src/core/compiler/compiler.js:126:43)
    at Compiler.execute.$__export._compileTemplate (http://localhost:8080/angular2/src/core/compiler/compiler.js:164:53)
    at http://localhost:8080/angular2/src/core/compiler/compiler.js:154:29
    at Zone.run (http://localhost:8080/dist/es6-shim.js:29891:19)
application.js:109 Uncaught (in promise) TypeError: Cannot read property 'annotations' of undefined {stack: (...), message: "Cannot read property 'annotations' of undefined"}
```

**Comment by wannadream** [comment]
I ran into the same issue.

<code>
TypeError: Cannot read property 'annotations' of undefined

TypeError: Cannot read property 'annotations' of undefined
    at ReflectionCapabilities.execute.$__export.annotations (http://localhost:3456/quickstart/angular2/src/reflection/reflection_capabilities.js:81:40)
    at Reflector.execute.$__export.annotations (http://localhost:3456/quickstart/angular2/src/reflection/reflector.js:81:50)
    at DirectiveMetadataReader.execute.$__export.read (http://localhost:3456/quickstart/angular2/src/core/compiler/directive_metadata_reader.js:31:41)
    at http://localhost:3456/quickstart/angular2/src/core/compiler/compiler.js:127:35
    at Array.map (native)
    at Function.execute.$__export.map (http://localhost:3456/quickstart/angular2/src/facade/collection.js:172:26)
    at Compiler.execute.$__export.createSteps (http://localhost:3456/quickstart/angular2/src/core/compiler/compiler.js:126:43)
    at Compiler.execute.$__export._compileTemplate (http://localhost:3456/quickstart/angular2/src/core/compiler/compiler.js:164:53)
    at http://localhost:3456/quickstart/angular2/src/core/compiler/compiler.js:154:29
    at Zone.run (http://localhost:3456/quickstart/dist/es6-shim.js:29889:19)
application.js:255 Uncaught (in promise) TypeError: Cannot read property 'annotations' of undefined {stack: (...), message: "Cannot read property 'annotations' of undefined"}message: "Cannot read property 'annotations' of undefined"stack: (...)get stack: function () { [native code] }set stack: function () { [native code] }__proto__: Error
</code>

**Comment by dicoy** [comment]
same here: todoApp

**Comment by vicb** [comment]
That's probably because an error in a View's directives, there is a PR pending on angular.
see https://github.com/angular/angular/pull/1917

**Comment by artdias90** [comment]
Same problem

**Comment by Vallerious** [comment]
Same here

**Comment by dwilt** [comment]
+1

**Comment by wardbell** [comment]
Closed. Obsolete.

**Comment by FredLoney** [comment]
@wardbell this should be reopened. The error is an uncaught exception that occurs in several situations. For me, it occurs when a route is defined without a component declaration. The main issue is that the error message is useless and difficult to trace. In my case, it indirectly results from the resolve.ts code:

`_ => resolver.resolveComponent(<any>node.value.component).then(factory => {
          node.value._resolvedComponentFactory = factory;` 

which cascades as an error in a wholly different call stack. The assignment above should be guarded by checking for a `factory` value and issuing the appropriate error. In addition, the route should be validated beforehand. Furthermore, there should be a separate task to ensure that the (forthcoming?) route documentation specifically states what is required.

This issue and its response are symptomatic of a pervasive cavalier disregard for defensive programming in the Angular development process. I appreciate how developers under pressure let issues like this slide, but technical leads and QA should know better. Code review, anyone? Unfortunately, defensive programming must be integral to the development process and is difficult to address after initial  functional goals are achieved.

**Comment by filipesilva** [comment]
@FredLoney the issue you are referring to should be reported in https://github.com/angular/angular rather than in the quickstart. 

The quickstart repo aims to provide a barebones development environment and is not the place to report issues with Angular 2 itself.

**Comment by FredLoney** [comment]
Oops, I thought all Angular modules shared a common issues list.

Filed as Angular Issue [#10168](https://github.com/angular/angular/issues/10168)
